src/animals/Crocodile.ts

```diff
@@ -11,45 +11,19 @@ export default class Crocodile extends Reptile {
     }
     this.length = length;
   }
+  
+  
+  giveBirth(): [string, number, boolean,string][] {
+    const crocodilesTuples = super.giveBirth();
+    for (var crocodileTuple of crocodilesTuples) {
 
-  giveBirth(): void {
-    if (this.isMale) {
-      console.log(
-        `Only Females can give birth and ${this.name} is a proud male ${this.constructor.name}`
-      );
-    } else if (this.home === undefined) {
-      console.log(
-        `${this.name} can't give birth because she doesn't have a home to meet a male`
-      );
-    } else {
-      //Giving birth for reptiles creates 8 reptiles also of random gender, but always at least 2 males and 1 female.
-      this.timesGivenBirth += 1;
-      for (let i = 1; i <= 5; i++) {
-        const num = Math.floor(Math.random() * (2 - 1 + 1)) + 1;
-        const name: string =
-          this.name + "-Baby-" + this.timesGivenBirth + "-" + i;
-        const gender: boolean = num === 1;
-        const age: number = 0;
-        const babyCroc = new Crocodile(name, age, gender, 10);
-        families.forEach((fam) => {
-          if (fam.name === this.home) {
-            fam.addAnimal(babyCroc, true);
-          }
-        });
-      }
-      for (let i = 6; i <= 8; i++) {
-        const name: string =
-          this.name + "-Baby-" + this.timesGivenBirth + "-" + i;
-        const gender: boolean = i % 2 != 0;
-        const age: number = 0;
-        const babyCroc = new Crocodile(name, age, gender, 10);
-        families.forEach((fam) => {
-          if (fam.name === this.home) {
-            fam.addAnimal(babyCroc, true);
-          }
-        });
-      }
-      console.log(`${this.name} lays eggs`);
+      const babySnake = new Crocodile(crocodileTuple[0], 0, crocodileTuple[2], 10);
+      families.forEach((family) => {
+        if (family.name === this.home) {
+          family.addAnimal(babySnake, true);
+        }
+      });
     }
+    return [["", 0, false,""]];
```

Comment:
this return statement is confusing, what is this supposed to return?

Reply:
nothing. The super method returns array of Tuples, so the method in the child class have to return the same kind of info. This info is never used. Couldn't think of better way

Reply:
you could specify the return type of both methods to be a union `AnimalCandidate[] | void` where AnimalCandidate looks like this: 
```
type AnimalCandidate = {
  name: string,
  isMale: boolean
}
```
 this way you will avoid the error but you will need to do a check before looping through the tuples like this 
```
 if (crocodilesTuples) {
      for (let crocodileTuple of crocodilesTuples) {
         const babySnake = new Crocodile(crocodileTuple.name, 0, crocodileTuple.isMale, 10);
         ...
        });
      }
    }
```
 to ensure that crocodilesTuples is an array. This is just one way that comes to mind but I think it would be even better to avoid this union return type somehow